### PR TITLE
fix: create a name-parent-owner unique index.

### LINKED ISSUE #195: BUG: user gets error only at the end of upload when parallel uploading the same file
On two occasions there is an error only after resumable upload of the the file finishes:
 - when the user uploads the same two files at the same time in the ui. (there can be two upload objects)
 - when the client  uploads the file via the gateway (directly) while another with same attributes already exists.

Due to rollback, it just wastes time for the user and doesn't harm the dbs.

### PR DESCRIPTION
**Pull request by AllKinds**
- fix #195 and (#62)
- fix: check that the schema index is ready before running the tests.
- fix: test that did not run before, and were not correct.

### CODE REVIEW

File: src/file/file.service.spec.ts [hunk 1]
```diff
@@ -98,25 +102,21 @@ describe('File Logic', () => {
...
-      });
+      const newUpload1: IUpload =
+        await UploadService.createUpload(testUpload.key, testUpload.bucket, 'name1', USER.id, null)
+          .should.eventually.exist;
+      const newUpload2: IUpload =
+        await UploadService.createUpload(KEY2, 'BUCKET2', 'name1', USER.id, null)
+          .should.eventually.be.rejectedWith(UniqueIndexExistsError);
     });
 
     it('should not throw an error when key already exists but bucket not', async () => {
```
Comment (AllKinds) [inline src/file/file.service.spec.ts:113]: Is this the wanted behavior @Shahar-Y ?

File: src/file/file.service.spec.ts [hunk 2]
```diff
@@ -49,11 +50,14 @@ const ExceedQuotaSize: number = 11 * GB;
 describe('File Logic', () => {
 
   before(async () => {
-    // Remove files from DB
+    await mongoose.connection.db.dropDatabase();
     const collections = ['files', 'uploads'];
     for (const i in collections) {
       mongoose.connection.db.createCollection(collections[i], (err) => { });
     }
```
Comment (Shahar-Y) [inline src/file/file.service.spec.ts:57]: What are these lines for? and why is quota not there if they are necessary?

File: src/upload/upload.model.ts [hunk 3]
```diff
@@ -60,10 +60,13 @@ export const uploadSchema: Schema = new Schema(
 
 uploadSchema.index({ key: 1, bucket: 1 }, { unique: true });
 
+uploadSchema.index({ name: 1, parent: 1, ownerID: 1 }, { unique: true, partialFilterExpression: { isUpdate : false } });
+
 // ******* SAME AS FILE *******//
 // handleE11000 is called when there is a duplicateKey Error
 const handleE11000 = function (error: MongoError, _: any, next: NextFunction) {
   if (error.name === 'MongoError' && error.code === 11000) {
+    console.log(error.message);
```
Comment (AllKinds) [inline src/upload/upload.model.ts:69]:
```suggestion
```

File: src/file/file.service.spec.ts (hunk 2 above)
Comment (AllKinds) [reply src/file/file.service.spec.ts:57]: You are the one who wrote that.

**Review by Shahar-Y — APPROVED** [review]

**Pull request merged**